urlcf7851e92b_Triploblastic_CSCD350_FinalProject_tgz-pJ8-MazeGame_MazeJ8/src/TriviaQuestions/DataBaseBuilder.java

```diff
@@ -0,0 +1,107 @@
+package TriviaQuestions;
+
+import java.io.File;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Random;
+import java.util.Scanner;
+
+/*For one-time use: creates new .db for MovieTrivia*/
+public class DataBaseBuilder {
+	private static Connection  c = null;
+	private static Statement  s = null;
+	private static int count = 0;
+	private static ArrayList<String> movies = new ArrayList();
+	
+	public static void main(String[] args) {
+		try {
+			connectToDatabase();
+			createTable();
+			insertTuples();
+		} catch (Exception e) {
+			System.out.println(e.toString());		
+		}
+	}//main
+	
+	public static void connectToDatabase() throws Exception {
+		Class.forName("org.sqlite.JDBC");
+		c = DriverManager.getConnection("jdbc:sqlite:MovieQuoteTrivia.db");
+		s = c.createStatement();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *s*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).